extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/KafkaConnectSources.java

```diff
@@ -87,33 +88,39 @@ public static StreamSource<SourceRecord> connect(@Nonnull Properties properties)
         }
 
         return Sources.streamFromProcessorWithWatermarks(name, true,
-                eventTimePolicy -> ProcessorMetaSupplier.randomMember(
-                        new ProcessorSupplier() {
-                            private transient ConnectorWrapper connectorWrapper;
-
-                            @Override
-                            public void init(@Nonnull Context context) {
-                                properties.put("tasks.max", Integer.toString(context.localParallelism()));
-                                this.connectorWrapper = new ConnectorWrapper(properties);
-                            }
-
-                            @Override
-                            public void close(@Nullable Throwable error) {
-                                if (connectorWrapper != null) {
-                                    connectorWrapper.stop();
-                                }
-                            }
-
-                            @Nonnull
-                            @Override
-                            public Collection<? extends Processor> get(int count) {
-                                return IntStream.range(0, count)
-                                        .mapToObj(i -> new ReadKafkaConnectP(connectorWrapper, eventTimePolicy))
-                                        .collect(Collectors.toList());
-                            }
+                eventTimePolicy -> ProcessorMetaSupplier.randomMember(processSupplier(properties, eventTimePolicy,
+                        projectionFn)));
+    }
 
-                        })
-        );
+    /**
+     * A generic Kafka Connect source provides ability to plug any Kafka
+     * Connect source for data ingestion to Jet pipelines.
+     * <p>
+     * You need to add the Kafka Connect connector JARs or a ZIP file
+     * contains the JARs as a job resource via {@link com.hazelcast.jet.config.JobConfig#addJar(URL)}
+     * or {@link com.hazelcast.jet.config.JobConfig#addJarsInZip(URL)}
+     * respectively.
+     * <p>
+     * After that you can use the Kafka Connect connector with the
+     * configuration parameters as you'd use it with Kafka. Hazelcast
+     * Jet will drive the Kafka Connect connector from the pipeline and
+     * the records will be available to your pipeline as {@link SourceRecord}s.
+     * <p>
+     * In case of a failure; this source keeps track of the source
+     * partition offsets, it will restore the partition offsets and
+     * resume the consumption from where it left off.
+     * <p>
+     * Hazelcast Jet will instantiate tasks on a random cluster member and use local parallelism for scaling.
+     * Property <code>tasks.max</code> is not allowed. Use {@link StreamStage#setLocalParallelism(int)} in the pipeline
+     * instead.
```

Comment:
Should we add a note that this might actually change in the future? We would like to get rid of this limitation.

Reply:
added https://github.com/hazelcast/hazelcast/pull/24203/commits/876cd54f8bf979f705aeb89d96dbbbd737e0dfd5